In [1]:
import pandas as pd


url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

Asimetría (Skewness): Investiguen cómo aplicar la función .skew() en la columna Age. Si el resultado es positivo o negativo, ¿qué significa físicamente sobre la edad de los pasajeros en el barco? (¿Había más jóvenes o más ancianos?).

In [2]:
skew_age = df['Age'].skew()
print(f"Asimetria de Age: {skew_age}")

Asimetria de Age: 0.38910778230082704


Aproximadamente +0.3891, lo que reprensenta una asimetria positiva \
Fisicamente significa que la mayor cantidad de los pasajeros eran pasajeros jovenes, mientras que habia pocos ancianos.

Curtosis (Kurtosis): Investiguen cómo aplicar la función .kurt() en la columna Fare. ¿Qué significa este número sobre el "grosor" de las colas de la distribución? ¿Confirma esto la existencia de los outliers que encontramos hoy en clase?

In [3]:
kurt_fare = df['Fare'].kurt()
print(f"Curtosis de Fare: {kurt_fare}")

Curtosis de Fare: 33.39814088089868


Aproximadamente +33.3981. \
Esto confirma que matematicamente la existencia de outlier extremos, la gran mayoria de los billetes costaban alrededor de las 30 libras, pero existen registros que sobrepasan mas de las 500 libras lo que nos lleva a tener un promedio que se dirije mas hacia un extremo de la cola.

Tablas Cruzadas (Cross-Tabulation): Investiguen la función pd.crosstab(). Generen una tabla cruzada entre la clase del pasajero (Pclass) y si sobrevivió (Survived). Esta técnica de sumarización matricial es el equivalente a una Tabla Dinámica en Excel. ¿Qué patrón logran visualizar rápidamente al cruzar ambas variables?

In [4]:
crosstab_counts = pd.crosstab(df['Pclass'], df['Survived'], margins=True)
print("Frecuencias absolutas:")
print(crosstab_counts)

crosstab_pct = pd.crosstab(df['Pclass'], df['Survived'], normalize='index') * 100
print("\nPorcentajes por clase (%):")
print(crosstab_pct.round(2))

Frecuencias absolutas:
Survived    0    1  All
Pclass                 
1          80  136  216
2          97   87  184
3         372  119  491
All       549  342  891

Porcentajes por clase (%):
Survived      0      1
Pclass                
1         37.04  62.96
2         52.72  47.28
3         75.76  24.24


Se logra observar que de la primera clase sobrevivio el 62.96% \
Mientras que en la segunda clase solamente sobrevivio el 47.28$ \ y la tercera clase solo sobrevivio el 24.24% de las personas. \
Lo que nos ayuda a entender que existe un nivel de supervivencia extra dentro del modelo que significa que la clase tambien influia dentro de las posibilidades de sobrevivir. Un pasajero de primera clase tenia casi 2.6 veces mas posibilidades de sobrevivir que una persona dentro de la tercera clase.